compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3224,11 +3235,11 @@ abstract class GenJSCode extends plugins.PluginComponent
     }
 
     /** Gen JS code for a call to AnyRef.synchronized */
-    private def genSynchronized(tree: Apply, isStat: Boolean): js.Tree = {
+    private def genSynchronized(tree: Tree, receiver: Tree, arg: Tree,
+        isStat: Boolean): js.Tree = {
```

Comment:
It's a common pattern in `GenJSCode`, though. We typically pass trees for context rather than positions, even when use the trees only for their `pos`.